src/bio2bel/sources/tfregulons.py

```diff
@@ -0,0 +1,138 @@
+        compression='zip',
+        usecols=['TF', 'target', 'effect', 'score', 'pubmedID_from_curated_resources'],
+    )
+    df.rename(
+        columns={
+            'pubmedID_from_curated_resources': 'pmids',
+            'TF': 'tf_hgnc_symbol',
+            'target': 'target_hgnc_symbol',
+        },
+        inplace=True,
+    )
+    df = df[df['score'].isin(set('ABC'))]  # there are also D and E
+
+    hgnc_name_to_id = get_name_id_mapping('hgnc')
+    df['tf_hgnc_id'] = df['tf_hgnc_symbol'].map(hgnc_name_to_id.get)
+    df['target_hgnc_id'] = df['target_hgnc_symbol'].map(hgnc_name_to_id.get)
+
+    tf_missing_id = df['tf_hgnc_id'].isna()
+    if tf_missing_id.any():
+        missing_tf_symbols = df.loc[tf_missing_id, 'tf_hgnc_symbol'].unique()
+        logger.warning(f'missing HGNC id for {len(missing_tf_symbols)} transcription factors')
+        df = df[~tf_missing_id]
+
+    target_missing_id = df['target_hgnc_id'].isna()
+    if target_missing_id.any():
+        missing_target_symbols = df.loc[target_missing_id, 'target_hgnc_symbol'].unique()
+        logger.warning(f'missing HGNC id for {len(missing_target_symbols)} targets')
+        df = df[~target_missing_id]
+
+    return df
+
+
+_df = None
+
+
+def get_hgnc_ids(graph: BELGraph) -> Set[str]:
+    """Get HGNC identifiers for nodes in the graph."""
+    return {
+        node.identifier
+        for node in graph
+        if isinstance(node, pybel.dsl.CentralDogma) and node.namespace.lower() == 'hgnc'
+    }
+
+
+def get_bel() -> BELGraph:
+    """Get the entirety of TFregulons as BEL."""
+    graph = BELGraph()
+    df = _get_df()
+    _add_rows(df, graph)
+    return graph
+
+
+def enrich_graph(graph: BELGraph) -> None:
+    """Enrich a graph with transcription factors effecting the genes/rnas/proteins in the graph."""
+    hgnc_ids = get_hgnc_ids(graph)
+    df = _get_df()
+    df = df[df['target_hgnc_id'].isin(hgnc_ids)]
+    _add_rows(df, graph)
+
+
+def _get_df() -> pd.DataFrame:
+    global _df
+    if _df is None:
+        _df = get_df()
+    return _df
+
+
+def _add_rows(df: pd.DataFrame, graph: BELGraph) -> None:
+    for _, row in df.iterrows():
+        tf_protein = pybel.dsl.Protein(
+            namespace='hgnc',
+            identifier=row['tf_hgnc_id'],
+            name=row['tf_hgnc_symbol'],
+        )
+        target_rna = pybel.dsl.Rna(
+            namespace='hgnc',
+            identifier=row['target_hgnc_id'],
+            name=row['target_hgnc_symbol'],
+        )
+        target_gene = target_rna.get_gene()
+
+        if 'pmids' in row:
+            citations = [pmid.strip() for pmid in row['pmids'].split(',')]
+        else:
+            citations = ['31340985']
+
+        evidence = 'From TFregulons'
+
+        for citation in citations:
+            graph.add_binds(
+                tf_protein,
+                target_gene,
+                citation=citation,
+                evidence=evidence,
+            )
+            graph.add_directly_increases(
```

Comment:
need to look into the original file and see what the `effect` column means, since it is in `{-1, 0, 1}`

Reply:
Solved in b389b9c